src/tangents.jl

```diff
@@ -410,18 +413,19 @@ It is an error for the zero element of the tangent space of `x` to be represente
 anything other than that which this function returns.
 """
 zero_tangent(x)
-@inline zero_tangent(::Union{Int8, Int16, Int32, Int64, Int128}) = NoTangent()
-@inline zero_tangent(x::IEEEFloat) = zero(x)
-@inline function zero_tangent(x::SimpleVector)
-    return map!(n -> zero_tangent(x[n]), Vector{Any}(undef, length(x)), eachindex(x))
-end
-@inline function zero_tangent(x::Array{P, N}) where {P, N}
-    return _map_if_assigned!(zero_tangent, Array{tangent_type(P), N}(undef, size(x)...), x)
+function zero_tangent(x::P) where {P}
+    if isbitstype(P)
+        return zero_tangent_internal(x)
+    end
+    return zero_tangent_internal(x, IdDict())
```

Comment:
This could be simplified to
```suggestion
    return isbitstype(P) ? zero_tangent_internal(x) : zero_tangent_internal(x, IdDict())
```